cloud/defaults.go

```diff
@@ -79,6 +79,11 @@ func GeneratePublicIPName(clusterName, hash string) string {
 	return fmt.Sprintf("%s-%s", clusterName, hash)
 }
 
+// GenerateNodeOutboundIPName generates a public IP name, based on the cluster name.
+func GenerateNodeOutboundIPName(clusterName string) string {
+	return fmt.Sprintf("%s-node-ip-outbound", clusterName)
```

Comment:
We may want to think about resource naming conventions: https://docs.microsoft.com/en-us/azure/cloud-adoption-framework/ready/azure-best-practices/naming-and-tagging.

See also https://kubernetes.slack.com/archives/CEX9HENG7/p1590166717280400

Reply:
Good point, thanks for pointing me to this resource (didn't know it existed). I will fix this name, and I opened https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/678 for broader auditing of resource naming.

Reply:
renamed to follow best practices but I kept the cluster name in there for now because it technically allows having multiple CAPZ clusters in one RG without conflict since public IPs are scoped to the RG. In the future, we should determine whether that's a real requirement and, if not, clearly document that each cluster has to be in its own RG (so we can remove the cluster-name / resource naming dependency).